File: algorithms/dutchNationalFlag/partition.cpp
#include <bits/stdc++.h>

using namespace std;

void dutch_national_flag(int array[], int len) {
  int low = 0;
  int high = len - 1;
  int mid = 0;

  while (mid <= high) {
    switch (array[mid]) {
      case 0:
        swap(array[low++], array[mid++]);
        break;
      case 1:
        mid++;
        break;
      case 2:
        swap(array[high--], array[mid]);
    }
  }
}

File: algorithms/dutchNationalFlag/partition.py
def dutch_national_flag(array: list, first_element: int, middle_element: int, final_element: int):
    low = 0
    high = len(array) - 1
    mid = 0

    while mid <= high:
        if array[mid] == first_element:
            array[mid], array[low] = array[low], array[mid]
            mid += 1
            low += 1
            continue
        if array[mid] == middle_element:
            mid += 1
            continue
        if array[mid] == final_element:
            array[mid], array[high] = array[high], array[mid]
            high -= 1
            continue



